src/PeerSelectState.h

```diff
@@ -151,11 +307,25 @@ class PeerSelector: public Dns::IpReceiver
     peer_t hit_type;
     ACLChecklist *acl_checklist;
 
-    typedef CbcPointer<PeerSelectionInitiator> Initiator;
-    Initiator initiator_; ///< recipient of the destinations we select; use interestedInitiator() to access
-
     const InstanceId<PeerSelector> id; ///< unique identification in worker log
 };
 
+template <class Key, typename FUNC> void
+PeerSelector::addGroup(CachePeersByKey<Key> &peers, FUNC getHierCode, const int groupId)
+{
+    typedef std::pair<Key, CachePeer *> KeyPeerPair;
+    std::sort(peers.begin(), peers.end(), [](const KeyPeerPair &a, const KeyPeerPair &b) { return a.first < b.first; });
```

Comment:
Well the std::sort is ascending by default (this is why in my original code used as is), however I believe it is better to use the compare function here, because the default comparator also checks the a.second < b.second. It does not causes problem in our case, but it is not needed.